src/counting_sort.rs

```diff
@@ -0,0 +1,95 @@
+// Implements http://rosettacode.org/wiki/Sorting_algorithms/Counting_sort
+#![allow(unstable)]
+
+fn counting_sort(array: &mut [i32]) {
```

Comment:
shouldn't min and max be known in advance? I.e. the signature per the rosettacode task should be `fn counting_sort(array: &mut [i32], min: i32, max: i32)`.
I would factor out the part from row 11 to row 21.